avro-converter/src/test/java/io/confluent/connect/avro/AvroDataTest.java

```diff
@@ -90,6 +93,109 @@
 
   // Connect -> Avro
 
+  @Test
+  public void testCyclicalAvroSchema() {
+    AvroDataConfig avroDataConfig = new AvroDataConfig.Builder()
+        .with(AvroDataConfig.CONNECT_META_DATA_CONFIG, false)
+        .with(AvroDataConfig.ENHANCED_AVRO_SCHEMA_SUPPORT_CONFIG, false)
+        .build();
+    AvroData listAvroData = new AvroData(avroDataConfig);
+    String linkedListAvroSchema =
+        "{\"type\": \"record\",\"name\": \"linked_list\",\"fields\" : "
+            + "[{\"name\": \"value\", \"type\": \"long\"},"
+            + "{\"name\": \"next\", \"type\": [\"null\", \"linked_list\"],\"default\" : null}]}";
+    org.apache.avro.Schema.Parser avroParser = new org.apache.avro.Schema.Parser();
+    org.apache.avro.Schema avroSchema = avroParser.parse(linkedListAvroSchema);
+
+    GenericRecord next = new GenericRecordBuilder(avroSchema)
+        .set("value", 2l)
+        .set("next", null)
+        .build();
+
+    GenericRecord headNode = new GenericRecordBuilder(avroSchema)
+        .set("value", 3l)
+        .set("next", next)
+        .build();
+
+    SchemaAndValue schema = listAvroData.toConnectData(avroSchema, headNode);
+    GenericRecord genericRecord = (GenericRecord) listAvroData.fromConnectData(
+        schema.schema(), schema.value());
+
+    assertEquals(headNode, genericRecord);
+
+  }
+
+  @Test
+  public void testGraph() {
```

Comment:
nit: just for clarity, indicating that this tests cyclic references via Array might help explain test coverage to a future reader